oak-run/src/main/java/org/apache/jackrabbit/oak/run/DataStoreCommand.java

```diff
@@ -155,21 +178,83 @@ private static boolean checkParameters(DataStoreOptions dataStoreOpts, Options o
     private void execute(NodeStoreFixture fixture,  DataStoreOptions dataStoreOpts, Options opts, Closer closer)
         throws Exception {
 
+        final BlobStoreOptions optionBean = opts.getOptionBean(BlobStoreOptions.class);
         try (Closer metricsCloser = Utils.createCloserWithShutdownHook()) {
             MetricsExporterFixture metricsExporterFixture =
                 MetricsExporterFixtureProvider.create(dataStoreOpts, fixture.getWhiteboard());
             metricsCloser.register(metricsExporterFixture);
 
-            MarkSweepGarbageCollector collector = getCollector(fixture, dataStoreOpts, opts, closer);
-            if (dataStoreOpts.checkConsistency()) {
-                long missing = collector.checkConsistency();
-                log.warn("Found {} missing blobs", missing);
+            if (dataStoreOpts.dumpRefs()) {
+                final File referencesTemp = File.createTempFile("traverseref", null, new File(opts.getTempDirectory()));
+                final BufferedWriter writer = Files.newWriter(referencesTemp, UTF_8);
+
+                boolean threw = true;
+                try {
+                    BlobReferenceRetriever retriever = getRetriever(fixture, dataStoreOpts, opts);
+
+                    retriever.collectReferences(new ReferenceCollector() {
+                        @Override
+                        public void addReference(String blobId, String nodeId) {
+                            try {
+                                Iterator<String> idIter = ((GarbageCollectableBlobStore) fixture.getBlobStore()).resolveChunks(blobId);
+
+                                while (idIter.hasNext()) {
+                                    String id = idIter.next();
+                                    final Joiner delimJoiner = Joiner.on(DELIM).skipNulls();
+                                    String line = VerboseIdLogger.encodeId(delimJoiner.join(id, escapeLineBreak(nodeId)),
+                                            optionBean.getBlobStoreType());
+                                    writeAsLine(writer, line, false);
+                                }
+                            } catch (Exception e) {
+                                throw new RuntimeException("Error in retrieving references", e);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid throwing raw exception types.](https://app.codacy.com/gh/oak-indexing/jackrabbit-oak/pullRequest?prid=5079914)

Reply:
can be ignored imo